src/main/java/core/basesyntax/MainApp.java

```diff
@@ -1,5 +1,15 @@
 package core.basesyntax;
 
-public class MainApp {
+public abstract class MainApp {
+    public static void main(String[] args) {
+        Machine bulldozer = new Bulldozer();
+        Machine truck = new Truck();
+        Machine excavator = new Excavator();
+        Machine[] machines = new Machine[]{excavator, bulldozer, truck};
```

Comment:
Make a note that you can call new Excavator(), new Bulldozer(), new Truck() explicitly in array initialization without temporary variables.

Reply:
okay